### PR TITLE
[FIX] gamification: email template for challenges

### PR DESCRIPTION
**Pull request by rrahir**
Description of the issue/feature this PR addresses:
New challenge report template for Gamification

Current behavior before PR:
no style in the mail what-soever.
Desired behavior after PR is merged:
stylish email with podium and user pictures

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/gamification/data/goal_base.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <odoo>
     <!-- Mail template is done in a NOUPDATE block
          so users can freely customize/delete them -->
-    <data noupdate="1">
+    <data noupdate="0">
```
Comment (Yenthe666) [inline addons/gamification/data/goal_base.xml]: Why oh why would you want to change a no update to an update? This'll break the whole logic of e-mail templates and updating your whole Odoo would revert possible made  changes.

File: addons/gamification/data/goal_base.xml [hunk 2]
```diff
@@ -20,76 +20,217 @@
...
+                            </table> 
+                        % endfor
                     % endif
+                </td></tr>
+            </table>
+            <!-- Footer -->
+            <table cellspacing="0" cellpadding="0" width="100%" style="padding: 15px;">
+                <tr>
+                    <td align="center" style="padding: 30px 15px;">
+                        <p style="font-size:12px;color:#AAAAAA">Odoo SA - Chaussée de Namur, 40, B-1367 Grand-Rosière</p>
```
Comment (Yenthe666) [inline addons/gamification/data/goal_base.xml]: Why hard code company data when you can get it from the company record?

File: addons/gamification/data/goal_base.xml (hunk 1 above)
Comment (rrahir) [inline addons/gamification/data/goal_base.xml]: Forgot to switch it back. thanks :)

File: addons/gamification/data/goal_base.xml [hunk 3]
```diff
@@ -20,76 +20,232 @@
...
+                                        style="font-weight:bold;"
+                                    % endif
+                                    >
+                                    <td style="padding: 20px 0;" align="left">
+                                        ${line['name']}
+                                        %if line['suffix'] or line['monetary']:
+                                            (${line['full_suffix']})
+                                        %endif
+                                    </td>
+                                    <td style="padding: 20px 0;" align="right">${"%.2f" %line['target']}
```
Comment (jem-odoo) [inline addons/gamification/data/goal_base.xml]: put a space after `%` in `%line[`

File: addons/gamification/data/goal_base.xml [hunk 4]
```diff
@@ -20,76 +20,232 @@
...
+                                    % if line['suffix']:
+                                        ${line['suffix']}
+                                    % endif
+                                    </td>
+                                    <td style="padding: 20px 0;" align="right">${"%.2f" %line['current']}
+                                    % if line['suffix']:
+                                        ${line['suffix']}
+                                    % endif
+                                    </td>
+                                    <td style="padding: 20px 0;font-size:25px;color:#9A6C8E;" align="right"><strong>${line['completeness']| int}%</strong></td>
```
Comment (jem-odoo) [inline addons/gamification/data/goal_base.xml]: What does the `| int` do ? Is it required ?

Comment (rrahir) [inline addons/gamification/data/goal_base.xml]: formats to Int. I guess it is similar to "int(line['completeness'])"

File: addons/gamification/data/goal_base.xml [hunk 5]
```diff
@@ -20,76 +20,232 @@
...
+                            <!-- Podium -->
+                            % if len(line['goals'])>2:
+                                <table cellspacing="0" cellpadding="0" width="100%" style="margin-top:10px;">
+                                    <tr><td style="padding:0 30px;">
+                                        <table cellspacing="0" cellpadding="0" width="100%">
+                                            <tr>
+                                                % set top_goals = [line['goals'][1], line['goals'][0], line['goals'][2]]
+                                                % for goal in top_goals:
+                                                    <td align="center" style="width:32%;"
+                                                        % if loop.index == 1:
```
Comment (rrahir) [inline addons/gamification/data/goal_base.xml]: I think it does not change a thing whether you define it inside or outside the tag. I followed the old template example.

**Pull request merged**